### ISSUE TITLE
Class constants

### ISSUE THREAD

**Issue by sagikazarmark** [umpirsky/list-generator#1]
Not sure if it makes sense for all lists, but a class with constants in it might be useful.

A usecase:

https://github.com/florianv/swap/issues/3

Examples:
- `Currency::EUR`
- `Locale:en_US`

It might be unnecessary complexity, also hard to generate. WDYT?

**Comment by umpirsky** [comment]
@sagikazarmark Thanks for the idea.

Well, how it would work?
- Would it be in the same `currency.php` file?
- Should it live inside root namespace?
- Do we really need locale constants?

And last question is, should it live inside https://github.com/umpirsky/currency-list repo? Maybe it deserves dedicated one?

**Comment by sagikazarmark** [comment]
> Would it be in the same currency.php file?

No, I think it should go into a separate file, and even directory, like src with autoloading.

> Should it live inside root namespace?

I think you could have a namespace like List\Currency

> Do we really need locale constants?

One of the reasons in the linked issue was autocompletion, so I think yes: everything you might want to type by hand and looks like a code, that should have a class with constants.

> And last question is,

I think it should live inside the repo where the relevant data is stored, I don't think it needs a dedicated repo.

**Comment by umpirsky** [comment]
If it lives inside same repository, then I would prefer to keep all php code inside single php file, as for other formats.

Now I am thinking, do we need classes at all?

Maybe global constants like `CURRENCY_LIST_ISO_EUR`?

**Comment by sagikazarmark** [comment]
I don't think its a good idea, for multiple reasons.
1. You can right now just require the file and it returns the array. What in case of a combined file?
2. Separation of concerns

Also, I think global namespace is evil. A class is bettwr IMO

**Comment by umpirsky** [comment]
@sagikazarmark I agree, I just think it is not suitable for list-generator project.

PHP format is here just because I am PHP developer myself. As you can see, no ruby, python, java...formats.

Expanding PHP format to more then one file is not what is expected from this project.

I am starting to think about `umpirsky/currency-list-php`.

**Comment by sagikazarmark** [comment]
Makes sense, I haven't looked at these lists from this point of view.

These repos (along with possible language-specific repos) could even go into their own organization/namespace. Their number might get high over time.

From this point of view, currency-list-php (or data-list/currency-php for example) would make sense.
